draft-rescorla-tls-esni.md

```diff
@@ -175,24 +175,18 @@ structure, defined below.
 
     struct {
         opaque label<0..2^8-1>;
-        KeyShareEntry share;
-    } ESNIKeyShareEntry;
-
-    struct {
-        ESNIKeyShareEntry keys<4..2^16-1>;
+        KeyShareEntry keys<4..2^16-1>;
```

Comment:
This makes it so that each label maps to a set of keys, which was not the original intent. We want there to be a 1-1 mapping between label and key, as that allows a server to abort a connection with illegal_parameter if there's a label mismatch, i.e., if a client offers a label that the server does not recognize.

Reply:
Please disregard! I did not read your comments in #19. The (label, group) => key mapping seems plausible. Though I don't think the NamedGroupList is necessary to prevent downgrade attacks.